blog.qmd

```diff
@@ -0,0 +1,329 @@
+keywords <- kwic(corpus, dict, window = 10) %>% as_tibble()
+```
+
+```{r}
+#| echo: false
+keywords %>% 
+  gt() %>% 
+  urbn_gt_theme() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) 
+```
+
+
+Finally, we can merge the results from the dictionary-based keyword in-context search back into our course-level data. This involves joining the keyword data with the original course data, followed by a series of data transformations. First, we split the `docname` document identifier column into separate `school` and `course` columns, which allows us to analyze the frequencies of WBL opportunities by school. Next, we construct a `sentence` column that encapsulates the matching keywords along with their context, and we format this text to ensure it is in sentence case. Lastly, we carry out a few more string manipulation steps on the `discipline` and `pattern` columns to enhance the readability of our data. It's also worth noting that the `pattern` column refers to the umbrella term used to describe the same type of WBL opportunity in our dictionary.
+
+```{r}
+courses_with_keywords <-
+  keywords %>% 
+  left_join(courses, by = c("docname" = "doc_id")) %>%
+  separate(docname, into = c("school", "course"), sep = "-", extra = 'merge'  ) %>% 
+  mutate(sentence = glue("{pre} {keyword} {post}") |> str_trim() |> str_to_sentence(),
+         discipline = str_remove(discipline, ".* - "),
+         pattern = str_to_title(pattern) |> str_replace_all('_', '-')) 
+  
+```
+
+```{r}
+#| echo: false
+courses_with_keywords %>% 
+  select(school, discipline, course, course_credits,  sentence, pattern) |>
+  arrange(school, pattern, discipline) %>% 
+  gt() %>% 
+  urbn_gt_theme() %>% 
+  opt_interactive(use_text_wrapping = FALSE, use_resizers = TRUE, use_compact_mode = TRUE) 
+```
+
+Having identified the keywords in the course descriptions, we now turn our attention to quantifying the prevalence of different types of WBL opportunities across schools. To do this, we group our data by school and pattern (which represents the type of WBL opportunity), count the number of occurrences, and then reshape our data so that each type of WBL opportunity becomes a separate column. We also add a 'total' row that sums up the counts across all schools for each type of opportunity.
+
+```{r}
+freq_table <-
+  courses_with_keywords %>%
+  group_by(school, pattern) %>%
+  count() %>%
+  group_by(pattern) %>%
+  bind_rows(summarise(.,
+                      across(where(is.numeric), sum),
+                      across(where(is.character), ~ "total"))) %>% 
+  ungroup() %>%
+  pivot_wider(names_from = pattern, values_from = n) %>%
+  mutate(across(where(is.numeric), ~ replace_na(.x, 0))) %>% 
+  slice(n(), 1:n()-1)
+```
+
+```{r}
+#| echo: false
+freq_table %>%
+  rename_with(~str_to_title(.x), everything()) %>% 
+  gt() %>% 
+  urbn_gt_theme() %>% 
+  tab_header(title = "Frequency of Work-Based Learning Opportunities by School")
+```
```

Comment:
This table looks so good!!

Reply:
I wrote it at the top of my .qmd file! I literally copy and paste this across projects so i'd be interested in discussing how to make it more available for folks if it's useful.